x-pack/plugins/saved_objects_tagging/public/management/components/table.tsx

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useRef, useEffect, FC } from 'react';
+import { EuiInMemoryTable, EuiLink } from '@elastic/eui';
+import { Action as EuiTableAction } from '@elastic/eui/src/components/basic_table/action_types';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { TagsCapabilities, TagWithRelations } from '../../../common';
+import { TagBadge } from '../../components';
+
+interface TagTableProps {
+  loading: boolean;
+  capabilities: TagsCapabilities;
+  tags: TagWithRelations[];
+  selectedTags: TagWithRelations[];
+  onSelectionChange: (selection: TagWithRelations[]) => void;
+  onEdit: (tag: TagWithRelations) => void;
+  onDelete: (tag: TagWithRelations) => void;
+  getTagRelationUrl: (tag: TagWithRelations) => string;
+  onShowRelations: (tag: TagWithRelations) => void;
+}
+
+const tablePagination = {
+  initialPageSize: 20,
+  pageSizeOptions: [5, 10, 20, 50],
+};
+
+const sorting = {
+  sort: {
+    field: 'name',
+    direction: 'asc' as const,
+  },
+};
+
+export const isModifiedOrPrevented = (event: React.MouseEvent) =>
+  event.metaKey || event.altKey || event.ctrlKey || event.shiftKey || event.defaultPrevented;
+
+export const TagTable: FC<TagTableProps> = ({
+  loading,
+  capabilities,
+  tags,
+  onSelectionChange,
+  selectedTags,
+  onEdit,
+  onDelete,
+  onShowRelations,
+  getTagRelationUrl,
+}) => {
+  const tableRef = useRef<EuiInMemoryTable<any>>();
+
+  useEffect(() => {
+    if (tableRef.current) {
+      tableRef.current.setSelection(selectedTags);
+    }
+  }, [selectedTags]);
+
+  const actions: Array<EuiTableAction<TagWithRelations>> = [];
+  if (capabilities.edit) {
+    actions.push({
+      name: i18n.translate('xpack.savedObjectsTagging.management.table.actions.edit.title', {
+        defaultMessage: 'Edit',
+      }),
+      description: i18n.translate(
+        'xpack.savedObjectsTagging.management.table.actions.edit.description',
+        {
+          defaultMessage: 'Edit this tag',
+        }
+      ),
+      type: 'icon',
+      icon: 'pencil',
+      onClick: (object: TagWithRelations) => onEdit(object),
+      'data-test-subj': 'tagsTableAction-edit',
+    });
+  }
+  if (capabilities.delete) {
+    actions.push({
```

Comment:
Same as above

Reply:
Resolving, will address both after discussion in previous comment.